Emby.Server.Implementations/HttpServer/WebSocketConnection.cs

```diff
@@ -174,7 +172,14 @@ private void OnReceiveInternal(string message)
                     Connection = this
                 };
 
-                OnReceive(info);
+                if (info.MessageType.Equals("KeepAlive", StringComparison.Ordinal))
```

Comment:
I think strict checks are a good idea. It doesn't help to be vague with the API functionality, we should clearly state exactly what works and what doesn't work.

Reply:
BTW, I'm afraid that all this `KeepAlive` part is going to be removed by using the same functionality that the protocol already provides.

I haven't tested it yet, but adding the following to `Jellyfin.Server/Startup.cs` should handle things without the need of this watcher or additional client code either.
```
app.UseWebSockets(new WebSocketOptions
{
   KeepAliveInterval = TimeSpan.FromSeconds(20),
});
```

The 20 seconds can be read from some configuration somewhere.

Should I test this, to avoid clutter code now, or leave the change for another PR?

Reply:
This code is needed because we want to handle dead websockets here.
Also docs say that default value of keepalive is 2 minutes, but I don't think it really works...

So let's leave current code for now.

Reply:
Agreed.